From 878c81f02e9cabbc990028b2cf87442813eb39f5 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Mon, 1 May 2017 15:18:05 -0400
Subject: [PATCH 16/22] show_errors(): make the useful part here not be dead
 code.

Woops.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/efivar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/efivar.c b/src/efivar.c
index cbefe8b..38d19e1 100644
--- a/src/efivar.c
+++ b/src/efivar.c
@@ -61,7 +61,7 @@ static int verbose_errors = 0;
 static void
 show_errors(void)
 {
-	int rc = 0;
+	int rc = 1;
 
 	if (!verbose_errors)
 		return;
-- 
2.12.2

